pkg/helm/v3/dependency.go

```diff
@@ -1,12 +1,18 @@
 package v3
 
 import (
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
 	"helm.sh/helm/v3/pkg/downloader"
 
 	"github.com/fluxcd/helm-operator/pkg/utils"
 )
 
 func (h *HelmV3) DependencyUpdate(chartPath string) error {
+	defer garbageCollect(repositoryCache, time.Second * 120)
```

Comment:
This is cleaning up files from _previous_ runs, so there's no reason to wait until after this run (it's not going to pick up any more files). I would remove the `defer`, and put a comment explaining why this is a convenient place to call it.